console/src/components/Services/Data/Schema/ConnectDatabase.tsx

```diff
@@ -0,0 +1,769 @@
+  }
+  if (!port) {
+    missingFields.push('port');
+  }
+  if (!username) {
+    missingFields.push('username');
+  }
+  if (!database) {
+    missingFields.push('database');
+  }
+
+  return `The following fields are required: ${missingFields
+    .slice(0, missingFields.length - 1)
+    .join(', ')} and ${missingFields[missingFields.length - 1]}.`;
+};
+
+const ConnectDatabase: React.FC<ConnectDatabaseProps> = props => {
+  const connectDBReducer = (
+    state: ConnectDBState,
+    action: { type: string; data: any }
+  ): ConnectDBState => {
+    switch (action.type) {
+      case UPDATE_DISPLAY_NAME:
+        return {
+          ...state,
+          displayName: action.data,
+        };
+      case UPDATE_DB_DRIVER:
+        return {
+          ...state,
+          dbType: action.data,
+        };
+      case UPDATE_DB_URL:
+        return {
+          ...state,
+          databaseURLState: {
+            dbURL: action.data,
+          },
+        };
+      case UPDATE_DB_URL_ENV_VAR:
+        return {
+          ...state,
+          envVarURLState: {
+            envVarURL: action.data,
+          },
+        };
+      case UPDATE_DB_HOST:
+        return {
+          ...state,
+          connectionParamState: {
+            ...state.connectionParamState,
+            host: action.data,
+          },
+        };
+      case UPDATE_DB_PORT:
+        return {
+          ...state,
+          connectionParamState: {
+            ...state.connectionParamState,
+            port: action.data,
+          },
+        };
+      case UPDATE_DB_USERNAME:
+        return {
+          ...state,
+          connectionParamState: {
+            ...state.connectionParamState,
+            username: action.data,
+          },
+        };
+      case UPDATE_DB_PASSWORD:
+        return {
+          ...state,
+          connectionParamState: {
+            ...state.connectionParamState,
+            password: action.data,
+          },
+        };
+      case UDPATE_DB_DATABASE_NAME:
+        return {
+          ...state,
+          connectionParamState: {
+            ...state.connectionParamState,
+            database: action.data,
+          },
+        };
+      case RESET_INPUT_STATE:
+        return {
+          ...defaultState,
+        };
+      case UPDATE_MAX_CONNECTIONS:
+        return {
+          ...state,
+          connectionSettings: {
+            ...state.connectionSettings,
+            max_connections: parseInt(action.data.trim(), 10),
+          },
+        };
+      case UPDATE_RETRIES:
+        return {
+          ...state,
+          connectionSettings: {
+            ...state.connectionSettings,
+            retries: parseInt(action.data.trim(), 10),
+          },
+        };
+      case UDPATE_IDLE_TIMEOUT:
+        return {
+          ...state,
+          connectionSettings: {
+            ...state.connectionSettings,
+            idle_timeout: parseInt(action.data.trim(), 10),
+          },
+        };
```

Comment:
Comment is specific towards L209, L217, and L225:
Can use a small helper for the conversion